general/administration/hardware-acceleration.md

```diff
@@ -6,7 +6,31 @@ title: Hardware Acceleration
 
 # Hardware Acceleration
 
-Jellyfin supports hardware acceleration of video encoding/decoding/transcoding using FFMpeg. It supports multiple acceleration types, including  AMD AMF, Intel Quick Sync, OpenMax OMX, nVidia NVENC, Intel/AMD VAAPI, and others.
+Jellyfin supports hardware acceleration of video encoding/decoding using FFMpeg. FFMpeg can support multiple hardware acceleration implementations like Intel Quicksync (QSV), AMD AMF, OpenMax OMX, nVidia NVENC/NVDEC through the Video Acceleration API (VAAPI), and others.
+
+OS | Recommended HW Acceleration
+------------ | -------------
+Linux/GNU | VAAPI (recommended), NVENC
```

Comment:
I think QSV also exists on Linux, and VAAPI (which exposes same hardware but via another interface) is not available on some older CPUs while QSV is.

Reply:
I'm not too familiar with HW Acceleration, but I am good at digging. There might be further commits to adjust the page in the future to correct information on specific hardware. Here's the research I have found so far for everything. For Linux, QSV will use libmfx library (see note at bottom). VAAPI will use the libva library.  The Debian link I added shows a list of supported HW for VA-API, we could incorporate this into the guide. 

https://github.com/Intel-FFmpeg-Plugin/Intel_FFmpeg_plugins/wiki 
"QSV transcode acceleration on Linux systems are enabled from FFMPEG 2.8 and forward"

AMF is supported on Linux but x265 coding is not. 
https://www.phoronix.com/scan.php?page=news_item&px=FFmpeg-AMD-AMF-Vulkan

VAAPI driver for Intel G45 & HD Graphics family uses the i965 va driver. 
https://packages.debian.org/sid/i965-va-driver

On Ubuntu, the guide for setting up QSV is actually just using VAAPI.  
https://wiki.ubuntu.com/IntelQuickSyncVideo 

ffmpeg page for QSV states only Kaby Lake can do HEVC. 
"Encode a 1080p HEVC main10 profile (supported since Kaby Lake platform)"
http://trac.ffmpeg.org/wiki/Hardware/QuickSync

As always, the Arch Wiki lol. It does not mention MSDK or libmfx. 
https://wiki.archlinux.org/index.php/Hardware_video_acceleration

Information on MSDK and libmfx
https://software.intel.com/en-us/articles/quick-start-on-integrating-ffmpeg-libraries

Shows a nice breakdown for ffmpeg
https://01.org/linuxmedia

Here is a note on the Ubuntu page about libmfx (MSDK)
```
Future Approach #2: Media SDK ("MSDK")
(This is the approach that the FFmpeg documentation rather confusingly refers to as "Intel QSV")

Intel's Media SDK is the company's professional offering with the aim of providing the highest performance and the most video features all accelerated on Intel chips.

It sounds great. You can download the SDK for free, and they have also published a first open source release of it too. But there are hidden reasons why we can't use any of this yet in Ubuntu. The problem is that pieces are missing. Both of them have a dependency on a custom proprietary fork of LibVA (VA-API). While this "missing" code is actually provided in the commercial download, it requires that you:

Overwrite official Ubuntu files, possibly breaking your system; and
Recompile all your video players yourself; and
Don't redistribute what you've made.
That won't work for us right now. And we don't recommend you try. It's very complicated and can easily break your Ubuntu installation.
```

I'm not sure how far and how much information the JF docs should contain. I have been getting a lot of push-back for things I wouldn't think would be an issue for the sake of minimalism and I'm having PRs sit in purgatory for it.

Reply:
Wow that is a lot of research, thank you for going through all the work to improve the hardware acceleration page!